bundles/model/org.eclipse.smarthome.model.thing.tests/src/test/groovy/org/eclipse/smarthome/model/thing/test/hue/GenericThingProviderTest2.groovy

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.model.thing.test.hue;
+
+import static org.hamcrest.CoreMatchers.*
+import static org.junit.Assert.assertThat
+
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory
+import org.eclipse.smarthome.model.core.ModelRepository;
+import org.eclipse.smarthome.test.OSGiTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.osgi.service.component.ComponentContext
+
+class GenericThingProviderTest2 extends OSGiTest{
+
+    private final static String TESTMODEL_NAME = "testModelX.things"
+
+    ModelRepository modelRepository;
+    ThingRegistry thingRegistry;
+
+    @Before
+    public void setUp() {
+        thingRegistry = getService ThingRegistry
+        assertThat thingRegistry, is(notNullValue())
+        modelRepository = getService ModelRepository
+        assertThat modelRepository, is(notNullValue())
+        modelRepository.removeModel(TESTMODEL_NAME)
+    }
+
+    @After
+    public void tearDown() {
+        modelRepository.removeModel(TESTMODEL_NAME);
+    }
+
+    @Test
+    public void 'assert that things are created and not removed on adding_removing triggerHandlerFactories'() {
+        def componentContextMock = [
+            getBundleContext: {getBundleContext()}
+        ] as ComponentContext
+
+        def hueThingHandlerFactory = new TestHueThingHandlerFactoryX(componentContextMock)
+
+        def things = thingRegistry.getAll()
+        assertThat things.size(), is(0)
+
+        String model =
+                '''
+			Bridge Xhue:Xbridge:myBridge [ XipAddress = "1.2.3.4", XuserName = "123" ] {
```

Comment:
Aaaaaahhh, TABS!!! 😱

Reply:
auto-formatter refuses to make all good, sorry :-(